packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -257,36 +257,47 @@ function translateTypeAnnotation(
       }
     }
     case 'TSTypeLiteral': {
-      const objectTypeAnnotation = {
-        type: 'ObjectTypeAnnotation',
-        // $FlowFixMe[missing-type-arg]
-        properties: (typeAnnotation.members: Array<$FlowFixMe>)
-          .map<?NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>>(
-            property => {
-              return tryParse(() => {
-                return parseObjectProperty(
-                  property,
-                  hasteModuleName,
-                  types,
-                  aliasMap,
-                  tryParse,
-                  cxxOnly,
-                  nullable,
-                  translateTypeAnnotation,
-                  parser,
-                );
-              });
-            },
-          )
-          .filter(Boolean),
-      };
-
-      return typeAliasResolution(
-        typeAliasResolutionStatus,
-        objectTypeAnnotation,
-        aliasMap,
-        nullable,
-      );
+      // if there is TSIndexSignature, then it is a dictionary
+      if (
+        typeAnnotation.members &&
+        typeAnnotation.members.filter(
+          member => member.type === 'TSIndexSignature',
+        ).length > 0
+      ) {
+        // no need to do further checking
+        return emitObject(nullable);
+      } else {
+        const objectTypeAnnotation = {
+          type: 'ObjectTypeAnnotation',
+          // $FlowFixMe[missing-type-arg]
+          properties: (typeAnnotation.members: Array<$FlowFixMe>)
+            .map<?NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>>(
+              property => {
+                return tryParse(() => {
+                  return parseObjectProperty(
+                    property,
+                    hasteModuleName,
+                    types,
+                    aliasMap,
+                    tryParse,
+                    cxxOnly,
+                    nullable,
+                    translateTypeAnnotation,
+                    parser,
+                  );
+                });
+              },
+            )
+            .filter(Boolean),
+        };
+
+        return typeAliasResolution(
+          typeAliasResolutionStatus,
+          objectTypeAnnotation,
+          aliasMap,
+          nullable,
+        );
+      }
```

Comment:
This block of code looks very similar to lines from 252 to 295 of `parsers/flow/modules/index.js`. Do you think we can unify them in `parsers-commons`, perhaps, adding the proper changes to the `Parser` interface and to the Parsers?

Reply:
Actually it is just two function call, by moving into `Parser`, code here still need to pass a huge amount of arguments to that function, it makes almost no differences after `yarn prettier`, so I prefer just keep them here.